modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1287,9 +1321,37 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
 
         if (openApiNullable) {
             if (Boolean.FALSE.equals(property.required) && Boolean.TRUE.equals(property.isNullable)) {
+                // Only add import when needed
                 model.imports.add("JsonNullable");
                 model.getVendorExtensions().put("x-jackson-optional-nullable-helpers", true);
             }
+            if (SpringCodegen.class.equals(this.getClass()) && Boolean.FALSE.equals(property.required) && Boolean.TRUE.equals(property.isNullable)) {
+                // Wrap dataType and defaults
+                property.isWrapped = true;
+                property.wrapperType = "JsonNullable";
+                property.wrapperFunc = "JsonNullable.of";
+                if (property.defaultValue != null) {
+                    property.defaultValue = "JsonNullable.of(" + property.defaultValue + ")";
+                } else {
+                    property.defaultValue = "JsonNullable.undefined()";
+                }
+            }
+        }
+        if (useOptional) {
+            if (Boolean.FALSE.equals(property.required) && Boolean.FALSE.equals(property.isNullable)
```

Comment:
I think this misses a case:

```
useOptional: true
openApiNullable: false

private Optional<String> message;
private Optional<String> nullableMessage
private String requiredMessage;
private String requiredNullableMessage;
```

I think this would skip `nullableMessage` because it's marked nullable (but the overall nullability feature is disabled)

Reply:
So I think this might still be a problem but I'm happy to iterate on this

Reply:
You mean it should be:

a) Boolean.FALSE.equals(property.required) && Boolean.FALSE.equals(property.isNullable)
or
b)  Boolean.FALSE.equals(property.required)  && Boolean.FALSE.equals(openApiNullable)

Reply:
Yes - but happy to follow up with that or merge into my PR dealing with Optionals in POJOs